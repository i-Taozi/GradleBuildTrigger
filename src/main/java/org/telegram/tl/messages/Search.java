/*
 *     This file is part of Telegram Server
 *     Copyright (C) 2015  Aykut Alparslan KOÃ‡
 *
 *     Telegram Server is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     Telegram Server is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package org.telegram.tl.messages;

import org.telegram.mtproto.ProtocolBuffer;
import org.telegram.tl.*;

public class Search extends TLObject {

    public static final int ID = 132772523;

    public TLInputPeer peer;
    public String q;
    public TLMessagesFilter filter;
    public int min_date;
    public int max_date;
    public int offset;
    public int max_id;
    public int limit;

    public Search() {
    }

    public Search(TLInputPeer peer, String q, TLMessagesFilter filter, int min_date, int max_date, int offset, int max_id, int limit){
        this.peer = peer;
        this.q = q;
        this.filter = filter;
        this.min_date = min_date;
        this.max_date = max_date;
        this.offset = offset;
        this.max_id = max_id;
        this.limit = limit;
    }

    @Override
    public void deserialize(ProtocolBuffer buffer) {
        peer = (TLInputPeer) buffer.readTLObject(APIContext.getInstance());
        q = buffer.readString();
        filter = (TLMessagesFilter) buffer.readTLObject(APIContext.getInstance());
        min_date = buffer.readInt();
        max_date = buffer.readInt();
        offset = buffer.readInt();
        max_id = buffer.readInt();
        limit = buffer.readInt();
    }

    @Override
    public ProtocolBuffer serialize() {
        ProtocolBuffer buffer = new ProtocolBuffer(32);
        serializeTo(buffer);
        return buffer;
    }

    @Override
    public void serializeTo(ProtocolBuffer buff) {
        buff.writeInt(getConstructor());
        buff.writeTLObject(peer);
        buff.writeString(q);
        buff.writeTLObject(filter);
        buff.writeInt(min_date);
        buff.writeInt(max_date);
        buff.writeInt(offset);
        buff.writeInt(max_id);
        buff.writeInt(limit);
    }

    public int getConstructor() {
        return ID;
    }
}