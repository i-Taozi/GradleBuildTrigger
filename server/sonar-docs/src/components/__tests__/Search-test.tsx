/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import lunr from 'lunr';
import * as React from 'react';
import { MarkdownRemark } from '../../@types/graphql-types';
import Search from '../Search';

jest.mock('lunr', () => ({
  __esModule: true,
  default: jest.fn(() => ({
    search: jest.fn(() => [
      {
        ref: 'lorem/origin',
        matchData: {
          metadata: {
            simply: {
              title: { position: [[19, 5]] },
              text: {
                position: [
                  [15, 6],
                  [28, 4]
                ],
                tokenContext: ['is simply dummy', 'simply dummy text']
              }
            }
          }
        }
      },
      {
        ref: 'foobar',
        matchData: {
          metadata: {
            simply: {
              title: { position: [[23, 4]] },
              text: {
                position: [
                  [111, 6],
                  [118, 4]
                ],
                tokenContext: ['keywords simply text']
              }
            }
          }
        }
      }
    ])
  }))
}));

function mockMarkdownRemark(override: Partial<MarkdownRemark>): MarkdownRemark {
  return {
    id: 'id',
    parent: null,
    children: null,
    internal: null,
    frontmatter: null,
    rawMarkdownBody: null,
    fileAbsolutePath: null,
    fields: null,
    html: null,
    htmlAst: null,
    excerpt: null,
    headings: null,
    timeToRead: null,
    tableOfContents: null,
    wordCount: null,
    ...override
  };
}

const pages = [
  mockMarkdownRemark({
    html:
      'Lorem Ipsum is simply dummy text of the printing and typesetting ' +
      "industry. Lorem Ipsum has been the industry's standard dummy text ever " +
      'since the 1500s, when an unknown printer took a galley of type and ' +
      'scrambled it to make a type specimen book.'
  }),
  mockMarkdownRemark({
    html:
      'Contrary to popular belief, Lorem Ipsum is not simply random text. ' +
      'It has roots in a piece of classical Latin literature from 45 BC, making' +
      ' it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney' +
      ' College in Virginia, looked up one of the more obscure Latin words.'
  }),
  mockMarkdownRemark({
    html:
      'Foobar is a universal variable understood to represent whatever is ' +
      'being discussed. Now we need some keywords: simply text.'
  })
];

it('should search', () => {
  const wrapper = shallow<Search>(
    <Search
      navigation={['lorem/index', 'lorem/origin', 'foobar']}
      pages={pages}
      onResultsChange={jest.fn()}
    />
  );
  wrapper.instance().handleChange({ currentTarget: { value: 'simply text+:' } } as any);
  expect(wrapper).toMatchSnapshot();
  expect(lunr).toBeCalled();
  expect(wrapper.instance().index).toBeDefined();
  expect((wrapper.instance().index as any).search).toBeCalledWith('simply~1 simply* text~1 text*');
});
