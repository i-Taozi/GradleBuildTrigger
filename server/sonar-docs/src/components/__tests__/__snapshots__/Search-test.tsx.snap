// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should search 1`] = `
<div
  className="search-container"
>
  <input
    aria-label="Search"
    className="search-input"
    onChange={[Function]}
    placeholder="Search..."
    type="search"
    value="simply text+:"
  />
  <button
    onClick={[Function]}
    type="button"
  >
    <ClearIcon
      size={8}
    />
  </button>
</div>
`;
